### DISCUSSION TITLE
Macro keypad

### DISCUSSION THREAD

**Discussion by macdroid53** [mnesarco/FreeCAD_Utils#12] [Q&A]
I'm thinking about one of these. https://www.tindie.com/products/dekunukem/duckypad-do-it-all-mechanical-macropad/

Is it likely to be a fit to get working with your utility...or do I even need a utility to use it...

**Comment by mnesarco** [comment]
I don't think you need any extension to configure your key bindings (shotcuts) in FreeCAD.

Just go to menu Tools -> Customize... -> Keyboard

I am not sure if it will be enough for what you want.

**Comment by mnesarco** [comment]
I am planning to add a dedicated FreeCAD layer to my Monster Keyboard, I will see if builtin shortcuts config is enough or not...

![monster](https://user-images.githubusercontent.com/1087317/220676097-65f59e63-98d3-4d8d-92f7-2b72b11d51d6.jpg)

**Comment by macdroid53** [comment]
What does the remote control server do?

**Comment by mnesarco** [comment]
Remote server serves a web app that you can access from any device connected to the same LAN/wifi and provides a remote UI for FreeCAD actions.

**Reply by mnesarco** [reply]
Here Is how it looks from my phone:

![WhatsApp Image 2023-02-22 at 11 11 15 AM](https://user-images.githubusercontent.com/1087317/220686376-42f9057b-d9d8-4614-af01-3e98931c0167.jpeg)

**Reply by mnesarco** [reply]
So if you put your phone near your keyboard, you have a dynamic FreeCAD touch keyboard. I also shows all macros as buttons.

**Reply by mnesarco** [reply]
![WhatsApp Image 2023-02-22 at 11 22 32 AM](https://user-images.githubusercontent.com/1087317/220689943-5fa7527b-e6da-43ff-852b-36504b3430be.jpeg)

**Comment by macdroid53** [comment]
Ahhh...that's interesting! So, theoretically an old table would work, right?

**Reply by mnesarco** [reply]
Sure
